File: packages/wasm/rs/web3api-wasm-rs/src/msgpack/format.rs
#[non_exhaustive]
pub struct Format;

impl Format {
    pub const ERROR: u8 = 0;
    pub const FOUR_LEAST_SIG_BITS_IN_BYTE: u8 = 0x0f;
    pub const FOUR_SIG_BITS_IN_BYTE: u8 = 0xf0;
    pub const POSITIVE_FIXINT: u8 = 0x00;
    pub const FIXMAP: u8 = 0x80;
    pub const FIXARRAY: u8 = 0x90;
    pub const FIXSTR: u8 = 0xa0;
    pub const NIL: u8 = 0xc0;
    pub const FALSE: u8 = 0xc2;
    pub const TRUE: u8 = 0xc3;
    pub const BIN8: u8 = 0xc4;
    pub const BIN16: u8 = 0xc5;
    pub const BIN32: u8 = 0xc6;
    pub const EXT8: u8 = 0xc7;
    pub const EXT16: u8 = 0xc8;
    pub const EXT32: u8 = 0xc9;
    pub const FLOAT32: u8 = 0xca;
    pub const FLOAT64: u8 = 0xcb;
    pub const UINT8: u8 = 0xcc;
    pub const UINT16: u8 = 0xcd;
    pub const UINT32: u8 = 0xce;
    pub const UINT64: u8 = 0xcf;
    pub const INT8: u8 = 0xd0;
    pub const INT16: u8 = 0xd1;
    pub const INT32: u8 = 0xd2;
    pub const INT64: u8 = 0xd3;
    pub const FIXEXT1: u8 = 0xd4;
    pub const FIXEXT2: u8 = 0xd5;
    pub const FIXEXT4: u8 = 0xd6;
    pub const FIXEXT8: u8 = 0xd7;
    pub const FIXEXT16: u8 = 0xd8;
    pub const STR8: u8 = 0xd9;
    pub const STR16: u8 = 0xda;
    pub const STR32: u8 = 0xdb;
    pub const ARRAY16: u8 = 0xdc;
    pub const ARRAY32: u8 = 0xdd;
    pub const MAP16: u8 = 0xde;
    pub const MAP32: u8 = 0xdf;
    pub const NEGATIVE_FIXINT: u8 = 0xe0;

    pub fn is_float_32(u: u8) -> bool {
        u == Self::FLOAT32
    }

    pub fn is_float_64(u: u8) -> bool {
        u == Self::FLOAT64
    }

    pub fn is_fixed_int(u: u8) -> bool {
        u >> 7 == 0
    }

    pub fn is_negative_fixed_int(u: u8) -> bool {
        (u & 0xe0) == Self::NEGATIVE_FIXINT
    }

    pub fn is_fixed_map(u: u8) -> bool {
        (u & 0xf0) == Self::FIXMAP
    }

    pub fn is_fixed_array(u: u8) -> bool {
        (u & 0xf0) == Self::FIXARRAY
    }

    pub fn is_fixed_string(u: u8) -> bool {
        (u & 0xe0) == Self::FIXSTR
    }

    pub fn is_nil(u: u8) -> bool {
        u == Self::NIL
    }
}

File: packages/wasm/as/assembly/msgpack/Format.ts
export const enum Format {
  ERROR = 0,
  FOUR_LEAST_SIG_BITS_IN_BYTE = 0x0f,
  FOUR_SIG_BITS_IN_BYTE = 0xf0,
  POSITIVE_FIXINT = 0x00,
  FIXMAP = 0x80,
  FIXARRAY = 0x90,
  FIXSTR = 0xa0,
  NIL = 0xc0,
  FALSE = 0xc2,
  TRUE = 0xc3,
  BIN8 = 0xc4,
  BIN16 = 0xc5,
  BIN32 = 0xc6,
  EXT8 = 0xc7,
  EXT16 = 0xc8,
  EXT32 = 0xc9,
  FLOAT32 = 0xca,
  FLOAT64 = 0xcb,
  UINT8 = 0xcc,
  UINT16 = 0xcd,
  UINT32 = 0xce,
  UINT64 = 0xcf,
  INT8 = 0xd0,
  INT16 = 0xd1,
  INT32 = 0xd2,
  INT64 = 0xd3,
  FIXEXT1 = 0xd4,
  FIXEXT2 = 0xd5,
  FIXEXT4 = 0xd6,
  FIXEXT8 = 0xd7,
  FIXEXT16 = 0xd8,
  STR8 = 0xd9,
  STR16 = 0xda,
  STR32 = 0xdb,
  ARRAY16 = 0xdc,
  ARRAY32 = 0xdd,
  MAP16 = 0xde,
  MAP32 = 0xdf,
  NEGATIVE_FIXINT = 0xe0,
}

export function isFloat32(u: u8): bool {
  return u == Format.FLOAT32;
}

export function isFloat64(u: u8): bool {
  return u == Format.FLOAT64;
}

export function isFixedInt(u: u8): bool {
  return u >> 7 == 0;
}

export function isNegativeFixedInt(u: u8): bool {
  return (u & 0xe0) == Format.NEGATIVE_FIXINT;
}

export function isFixedMap(u: u8): bool {
  return (u & 0xf0) == Format.FIXMAP;
}

export function isFixedArray(u: u8): bool {
  return (u & 0xf0) == Format.FIXARRAY;
}

export function isFixedString(u: u8): bool {
  return (u & 0xe0) == Format.FIXSTR;
}

export function isNil(u: u8): bool {
  return u == Format.NIL;
}
